test-case-reporting/test/testing_utils.ts

```diff
@@ -45,68 +45,41 @@ export async function fillDatabase(db: Database): Promise<void> {
     .returning('id');
 
   const jobIds: Array<number> = [];
+  let jobId;
 
-  let [jobId] = await db('jobs').insert({
-    'build_id': buildId,
-    'job_number': '12123434.1',
-    'test_suite_type': 'unit',
-  } as DB.Job);
-  jobIds.push(jobId);
+  for (let i = 1; i <= 3; i++) {
+    for (const suiteType of ['unit', 'integration']) {
+      [jobId] = await db('jobs').insert({
+        'build_id': buildId,
+        'job_number': `12123434.${i}`,
+        'test_suite_type': suiteType,
+      } as DB.Job);
 
-  [jobId] = await db('jobs').insert({
-    'build_id': buildId,
-    'job_number': '12123434.1',
-    'test_suite_type': 'integration',
-  } as DB.Job);
-  jobIds.push(jobId);
+      jobIds.push(jobId);
+    }
+  }
 
-  [jobId] = await db('jobs').insert({
-    'build_id': buildId,
-    'job_number': '12123434.2',
-    'test_suite_type': 'unit',
-  } as DB.Job);
-  jobIds.push(jobId);
+  const testCaseNames: Array<string> = ['case | 1', 'case | 2', 'case | 3'];
+  const testCases: Array<DB.TestCase> = testCaseNames.map(name => ({
+    id: md5(name),
+    name,
+  }));
+  const testCaseIds: Array<string> = testCases.map(({id}) => id);
 
-  [jobId] = await db('jobs').insert({
-    'build_id': buildId,
-    'job_number': '12123434.2',
-    'test_suite_type': 'integration',
-  } as DB.Job);
-  jobIds.push(jobId);
-
-  [jobId] = await db('jobs').insert({
-    'build_id': buildId,
-    'job_number': '12123434.3',
-    'test_suite_type': 'unit',
-  } as DB.Job);
-  jobIds.push(jobId);
-
-  const testCaseIds: Array<string> = await db('test_cases')
-    .insert([
-      {
-        id: md5('case | 1'),
-        name: 'case | 1',
-      },
-      {
-        id: md5('case | 2'),
-        name: 'case | 2',
-      },
-      {
-        id: md5('case | 3'),
-        name: 'case | 3',
-      },
-    ] as Array<DB.TestCase>)
-    .returning('id');
+  await db('test_cases').insert(testCases);
 
   const dbTestRuns: Array<DB.TestRun> = [];
 
-  jobIds.forEach((jobId: number) => {
-    testCaseIds.forEach((testCaseId: string) => {
+  const oneHourAgoMs = Date.now() - 60 * 60 * 1000;
+
+  jobIds.forEach((jobId, index) => {
+    testCaseIds.forEach(testCaseId => {
       dbTestRuns.push({
         'job_id': jobId,
         'test_case_id': testCaseId,
         status: 'PASS',
         'duration_ms': 4242,
+        timestamp: oneHourAgoMs + index,
```

Comment:
This feels hacky and I have a few thoughts for fixing it depending on how the test turns out. Leave this unresolved for now and I'll look at it once you've had a chance to address other comments

Reply:
Understood.

Reply:
Removed.